Detectors/MUON/MCH/Calibration/include/MCHCalibration/MCHChannelCalibrator.h

```diff
@@ -0,0 +1,132 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef MCH_CHANNEL_CALIBRATOR_H_
+#define MCH_CHANNEL_CALIBRATOR_H_
+
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include "MCHCalibration/PedestalDigit.h"
+#include "MCHCalibration/PedestalProcessor.h"
+
+#include <array>
+#include <boost/histogram.hpp>
+
+namespace o2
+{
+namespace mch
+{
+namespace calibration
+{
+
+class SampaChannelId
+{
+public:
+  SampaChannelId() = default;
+  SampaChannelId(uint32_t channelId): mChannelId(channelId) {}
+  SampaChannelId(uint16_t solarId, uint8_t dsId, uint8_t channel)
+  {
+    set(solarId, dsId, channel);
+  }
+
+  void set(uint16_t solarId, uint8_t dsId, uint8_t channel)
+  {
+    mChannelId = (static_cast<uint32_t>(solarId) << 16) +
+        (static_cast<uint32_t>(dsId) << 8) + channel;
+  }
+
+private:
+  uint32_t mChannelId{0};
+
+  ClassDefNV(SampaChannelId, 1); // class for MCH readout channel
+};
+
+class BadChannelsVector
```

Comment:
if your calibration object should contain nothing but the content of this vector, then it is indeed better to define ``SampaChannelId`` and ``BadChannelsVector`` in a single header in DataFormats, and use the latter as CCDB object.

Reply:
Done.